src/main/kotlin/org/quiltmc/community/modes/quilt/extensions/UtilityExtension.kt

```diff
@@ -104,6 +106,15 @@ class UtilityExtension : Extension() {
 
                 logger.info { "Thread created by ${owner.tag}" }
 
+                threads.set(
```

Comment:
We don't need to insert this for every new thread, we should only insert when something happens to modify the collection outside of the norm